frontend/express/public/javascripts/countly/vue/components/vis.js

```diff
@@ -1594,6 +1594,111 @@
                     </div>'
     }));
 
+    Vue.component("cly-map-picker", countlyVue.components.create({
+        components: {
+            'l-map': Vue2Leaflet.LMap,
+            'l-tile-layer': Vue2Leaflet.LTileLayer,
+            'l-marker': Vue2Leaflet.LMarker,
+            'l-circle': Vue2Leaflet.LCircle,
+        },
+        props: {
+            radius: {
+                type: Object,
+                required: false,
+                validator: function(value) {
+                    return value && (value.unit === 'km' || value.unit === 'mi') && value.value;
+                },
+                default: null
+            }
+        },
+        data: function() {
+            return {
+                markerCoordinates: null,
+                defaultCenterCoordinates: {lat: 48.66194284607008, lng: 8.964843750000002}, // Note: arbitrary coordinates used for map center when user location is not found
+                userCenterCoordinates: null,
+                tileFeed: 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
```

Comment:
This might be a problem for offline users. :slightly_smiling_face:

Reply:
Well, one way it so move that information to our server. Not sure how much is there though. In saying that, the implementation is the same as what we had, so not sure how the offline users were supported.